Source/WebCore/css/ComputedStyleExtractor.cpp

```diff
@@ -2820,6 +2821,53 @@ static inline bool isFlexOrGridItem(RenderObject* renderer)
     return box.isFlexItem() || box.isGridItem();
 }
 
+static Ref<CSSValue> valueForScrollTimelineAxis(const Vector<ScrollAxis>& axes)
+{
+    if (axes.isEmpty())
+        return CSSPrimitiveValue::create(CSSValueBlock);
+
+    CSSValueListBuilder list;
+    for (auto axis : axes)
+        list.append(createConvertingToCSSValueID(axis));
+    return CSSValueList::createCommaSeparated(WTFMove(list));
+}
+
+static Ref<CSSValue> valueForScrollTimelineName(const Vector<AtomString>& names)
+{
+    if (names.isEmpty())
+        return CSSPrimitiveValue::create(CSSValueNone);
+
+    CSSValueListBuilder list;
+    for (auto& name : names) {
+        if (name.isNull())
+            list.append(CSSPrimitiveValue::create(CSSValueNone));
+        else
+            list.append(CSSPrimitiveValue::createCustomIdent(name));
+    }
+    return CSSValueList::createCommaSeparated(WTFMove(list));
+}
+
+static Ref<CSSValue> scrollTimelineShorthandValue(const Vector<Ref<ScrollTimeline>>& timelines)
+{
+    if (timelines.isEmpty())
+        return CSSPrimitiveValue::create(CSSValueNone);
+
+    CSSValueListBuilder list;
+    for (auto& timeline : timelines) {
+        auto& name = timeline->name();
+        auto axis = timeline->axis();
+
+        ASSERT(!name.isNull());
+        auto nameCSSValue = CSSPrimitiveValue::createCustomIdent(name);
+
+        if (axis == ScrollAxis::Block)
+            list.append(WTFMove(nameCSSValue));
+        else
+            list.append(CSSValueList::createSpaceSeparated(nameCSSValue, createConvertingToCSSValueID(axis)));
```

Comment:
```suggestion
            list.append(CSSValuePair::createNonCoalescing(nameCSSValue, createConvertingToCSSValueID(axis)));
```
You can use CSSValuePair here.